app/javascript/Pages/Classes.jsx

```diff
@@ -1,13 +1,206 @@
-import React, { Component } from "react";
+import React, { useState, useEffect } from "react";
 import Layout from "./Layout";
-class Classes extends Component {
-  render() {
-    return (
-      <Layout current_user={this.props.current_user}>
-        <p>Hello, welcome to your first class!</p>
-      </Layout>
+import ButtonFilter from "./components/ButtonFilter";
+import { FaTimes } from "react-icons/fa";
+import PropTypes from "prop-types";
+
+const Classes = (props) => {
+  const [categories, setCategories] = useState([]);
+  const [skills, setSkills] = useState([]);
+  const [classes, setClasses] = useState([]);
+  const [categoryFilter, setCategoryFilter] = useState("");
+  const [skillFilter, setSkillFilter] = useState("");
+  const [difficultyFilter, setDifficultyFilter] = useState("");
+
+  const updateCategory = (category) => {
+    setCategoryFilter(category.id);
+    if (category) skillsPopup(category.skills);
+    else {
+      setSkills([]);
+    }
+    setSkillFilter("");
+  };
+
+  const fetchClasses = async () => {
+    const params = new URLSearchParams();
+
+    if (difficultyFilter) {
+      params.append("difficulty", difficultyFilter);
+    }
+    if (categoryFilter) {
+      params.append("category_id", categoryFilter);
+    }
+    if (skillFilter) {
+      params.append("skill_id", skillFilter);
+    }
+
+    let url = "/api/classes";
+
+    if (params.toString()) {
+      console.log(params.toString());
+      url += "?" + params.toString();
+    } else {
+      console.log("no parameters");
+    }
+
+    console.log(url);
+    let classes = [];
+    try {
+      const response = await fetch(url);
+      classes = await response.json();
+    } catch (error) {
+      console.log(error);
+      classes = [];
+    }
+
+    const components = classes.map((c) => {
+      return <p>{c.title}</p>;
+    });
+
+    setClasses(components);
+  };
+
+  const addCategoryButtons = async () => {
```

Comment:
**nitpick (non-blocking)**: this does not actually add the buttons right? a better name could be "generateCategoryButtons" or "getCategoryButtons"